examples/htmlcollector/execollector.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"bytes"
+	"io"
+	"mime/multipart"
+	"net/http"
+
+	"github.com/NextronSystems/jsonlog/thorlog/v3"
+	"github.com/NextronSystems/thor-plugin"
+)
+
+const sampleServerUrl = "http://localhost:8084/upload"
+
+func Init(config thor.Configuration, logger thor.Logger, actions thor.RegisterActions) {
+	actions.AddPostProcessingHook(uploadSample)
+	logger.Info("HTMLCollector plugin loaded!")
+}
+
+func uploadSample(logger thor.Logger, object thor.MatchedObject) {
+	file, isFile := object.Object.(*thorlog.File)
+
+	// Skip if the object is not a file
+	if !isFile {
+		return
+	}
+
+	// Select only EXE files
+	if file.MagicHeader != "EXE" {
+		return
+	}
+
+	body := &bytes.Buffer{}
```

Comment:
This copies the full file into RAM before writing it to the server. For an example, it's okay, but maybe we should add a comment about this.